### PR TITLE
ICU-22221 Fixes ICU demos with new cmemory.h

### PR DESCRIPTION
**Pull request by sven-oly**
This fixes run failures in demos by replacing cmemory.h with the latest version. It also updates the side bar and adds the sidebar to Segments.

### CODE REVIEW

File: apputil/data/icu.css [hunk 1]
```diff
@@ -18,6 +18,13 @@ body {
 .mainbody {
 	padding: 1em;
 }
+#main {
+	padding: 10px
+}
+
+/*
+ * A bit of space to right of left navigation
```
Comment (markusicu) [inline apputil/data/icu.css:26]:
```suggestion
 * A bit of space to the right of the left navigation bar.
```

**Review by srl295 — APPROVED** [review]
LGTM .  Maybe cmemory.h should be exported somewhere?  Maybe behind a required #define so it in't included accidentally?

**Comment by markusicu** [comment]
> Maybe cmemory.h should be exported somewhere? Maybe behind a required #define so it in't included accidentally?

I would
- either just include the internal cmemory.h
- or change the demo code to use standard memory management functions

... in a future PR.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Comment by srl295** [comment]
> > Maybe cmemory.h should be exported somewhere? Maybe behind a required #define so it in't included accidentally?
> 
> I would
> 
> * either just include the internal cmemory.h

It's a copy, but it's not always #include-able.  For example, if you get the headers and libraries via a standard package manager.
 
> * or change the demo code to use standard memory management functions

That would be good. Maybe reasonable with the current state of C++.

> ... in a future PR.

**Comment by markusicu** [comment]
https://unicode-org.atlassian.net/browse/ICU-22386 make icu-demos code not use cmemory.h

**Pull request merged**